### ISSUE TITLE
Time Filter accessibility refinement

### ISSUE THREAD

**Issue by zebdelk** [cedardevs/onestop#420]
- [ ] When date fields have default fields populated (like month & day set to Jan 1 when only given year), screen readers must receive alert that this change was performed.
  - the simplest solution here is to include that information in the placeholder
- [x] error message should be specific enough for a screen reader to know which date is wrong and why, since the visual checkmark is not used by the screen reader
- [x] clear button should have aria-label that specifies 'clear start and end date filters'
- [x] apply button should have aria-label that specifies 'apply start and end date filters'
- [x] use aria-placeholder, since screen readers are not reporting the input placeholder attribute.
  - 'Y Y Y Y' (YYYY does not read in as useful of a way, although it's possible screenreader users are used to that enough to parse what it means)
  - 'D D defaults to 1 if unset'
  - 'month defaults to January if unset'
- [x] label positioning. This will work best if the labels are directly before the field they apply to, and positioned so they appear the same visually. Each individual field can be a flex column, nested inside a flex row, for example. (Screen readers can and do use the label for attribute, but not consistently enough to skip this)
- [x] enter button should apply the filter on each field
- [ ] use aria validation attribute to indicate the element with the validation error
- [x] improve aria labels so that out of context (ie, jump to next form field) so that year has more information (possibly year - end date?)

**Comment by zebdelk** [comment]
Opted to skip communicating the default month and day information - there's nowhere sensible to present it that isn't more disruptive than assuming it's okay. If this is a problem, it will come up in later accessibility reviews.

**Comment by zebdelk** [comment]
Opted to skip aria validation attribute (aria-invalid and aria-errormessage), since they did not improve screenreader experience (alert is sufficient for now).

Note that this is in part because it's not really possible to set aria-invalid on the correct elements right now, and we'd need to restructure the filter to do so.
